msvc/sdk-build.targets

```diff
@@ -64,7 +64,15 @@
       <PackageFile Include="@(_WinMDFileNames->Distinct())" Condition="Exists('%(FullPath)')">
         <PackagePath>$(PackageContentBasePath)\%(Filename)%(Extension)</PackagePath>
       </PackageFile>
-    </ItemGroup>
+      
+      <!-- Package up .pri files associated with any .winmds -->
+      <_PriFileNames Include="$(ProjectPriFullPath)">
+      </_PriFileNames>
+      <PackageFile Include="@(_PriFileNames)" Condition="Exists('%(RootDir)%(Directory)%(Filename).winmd')">
```

Comment:
You aren't considering whats really going on here. Remember that this is in sdk-build.targets (and you are going to add something similar to the middleware targets). This needs to work generically. Things to consider:

1. You can't package a file that doesn't exist.

 Its imperative that when you add a PackageFile item, it exists. Currently you don't check at all if this exists. Its possible in a different scenario that on disk the pri file isn't at ProjectPriFullPath or that ProjectPriFullPath is defined but a Pri file isn't made. Perhaps in the scenario of a non AppContainerApplication project, no pri file is created.

2. The winmd file may not be located at %(RootDir)%(Directory)%(Filename).winmd

People can change where the winmd is output similar to changing their output directory. This combination is using metadata (the % ) off of each item from the expansion of _PriFileNames, so you are assuming that the pri file is *always* next to the winmd. That is true for UIKit.Xaml but is that true always? 

The precise condition that you are wanting to capture is "Only add a pri file if I have a pri file on disk and that this project produces a winmd"

the current condition is "Only add a pri file if there is a winmd file on disk in the same folder where I might write a pri file"

in the UIKit.xaml case this is equivalent but in the general case I would like to be more exact.

---
In reply to: [108994641](https://github.com/Microsoft/WinObjC/pull/2372#discussion_r108994641) [](ancestors = 108994641,108994152)